--- a/dns.c.orig	2010-10-19 05:46:03.000000000 +0200
+++ b/dns.c	2010-10-19 05:53:05.000000000 +0200
@@ -445,11 +445,11 @@
 AtomPtr
 rfc2732(AtomPtr name)
 {
-    char buf[38];
+    char buf[40]; /* 8*4 (hexdigits) + 7 (colons) + 1 ('\0') */
     int rc;
     AtomPtr a = NULL;
 
-    if(name->length < 38 && 
+    if(name->length < 40+2 && 
        name->string[0] == '[' && name->string[name->length - 1] == ']') {
         struct in6_addr in6a;
         memcpy(buf, name->string + 1, name->length - 2);
